src/player.js

```diff
@@ -130,53 +158,95 @@ class Player extends FakeEventTarget {
    * @static
    */
   static _defaultConfig(): Object {
-    return {};
+    return DefaultPlayerConfig;
   }
 
   /**
-   *
-   * @param {Object} config - The configuration of the player instance.
+   * Loads the configured plugins.
    * @private
    * @returns {void}
    */
-  _loadPlugins(config: Object): void {
-    let plugins = config.plugins;
+  _loadPlugins(): void {
+    let plugins = this._config.plugins;
     for (let name in plugins) {
       this._pluginManager.load(name, this, plugins[name]);
     }
   }
 
   /**
-   * Select the engine to create based on the given configured sources.
-   * @param {Object} config - The configuration of the player instance.
+   * Selects the engine to create based on a given configuration.
+   * @private
+   * @returns {boolean} - Whether a proper engine was found.
+   */
+  _selectEngine(): boolean {
+    let engineSelected = false;
+    if (this._config.sources) {
```

Comment:
We can always just do _selectEngineByPriority, cause we have the default setup.
Now just need to make sure the default config is deep cloned.